src/Schema/Building/SchemaBuilder.php

```diff
@@ -53,17 +54,25 @@ public function build(
     }
 
     /**
-     * @inheritdoc
+     * @param DocumentNode              $document
+     * @param ResolverRegistryInterface $resolverRegistry
+     * @param array $options
+     * @return BuildingContextInterface
+     * @throws BuildingException
+     * @throws InvalidArgumentException
      */
-    public function createContext(
+    protected function createContext(
         DocumentNode $document,
-        ResolverRegistryInterface $resolverRegistry
+        ResolverRegistryInterface $resolverRegistry,
+        array $options
     ): BuildingContextInterface {
         $info = $this->createInfo($document);
 
         $definitionBuilder = new DefinitionBuilder(
             $info->getTypeDefinitionMap(),
             $resolverRegistry,
+            $options['types'] ?? [],
+            $options['directives'] ?? [],
```

Comment:
Add support for passing `types` and `directives` with `$options` to the `DefinitionBuilder`.